sesman/sesman.c

```diff
@@ -187,6 +246,90 @@ static int sesman_listen_test(struct config_sesman *cfg)
 
     return rv;
 }
+
+/******************************************************************************/
+int
+sesman_close_all(void)
+{
+    int index;
+    struct sesman_con *sc;
+
+    LOG_DEVEL(LOG_LEVEL_TRACE, "sesman_close_all:");
+    trans_delete(g_list_trans);
+    for (index = 0; index < g_con_list->count; index++)
+    {
+        sc = (struct sesman_con *) list_get_item(g_con_list, index);
+        delete_connection(sc);
+    }
+    return 0;
+}
+
+/******************************************************************************/
+static int
+sesman_data_in(struct trans *self)
+{
+    int version;
+    int size;
+
+    if (self->extra_flags == 0)
+    {
+        in_uint32_be(self->in_s, version);
+        in_uint32_be(self->in_s, size);
+        if (size > self->in_s->size)
+        {
+            LOG(LOG_LEVEL_ERROR, "sesman_data_in: bad message size");
+            return 1;
+        }
+        self->header_size = size;
+        self->extra_flags = 1;
+    }
+    else
+    {
+        /* process message */
+        struct sesman_con *sc = (struct sesman_con *)self->callback_data;
+        self->in_s->p = self->in_s->data;
+        if (scp_process(self, sc->s) != SCP_SERVER_STATE_OK)
+        {
+            LOG(LOG_LEVEL_ERROR, "sesman_data_in: scp_process_msg failed");
+            return 1;
+        }
+        /* reset for next message */
+        self->header_size = 8;
+        self->extra_flags = 0;
+        init_stream(self->in_s, 0);
+    }
+    return 0;
+}
+
+/******************************************************************************/
+static int
+sesman_listen_conn_in(struct trans *self, struct trans *new_self)
+{
+    struct sesman_con *sc;
+    if (g_con_list->count >= 16)
```

Comment:
why a maximum of 16 connections to sesman? are the connections to sesman short lived or do they last the duration of the client rdp connection?

Reply:
The '16' is somewhat inherited. Your question is however, like most of your questions, very incisive.

At the moment, connections to sesman are short lived, and last for the duration of making a request and receiving a response.

This may, however, have to change in the future. There are two reasons I'm aware of for this:-
- To support reconnections, sesman is going to need to keep a copy of a secret which is shared with the RDP client. This secret is regenerated hourly - see [MS-RDPBCGR] 5.5 which implies either a permanent connection between sesman and the xrdp instance, or xrdp reconnecting to sesman at regular intervals.
- The `xrdp-sesadmin` utility should support more functionality that it currently does. Implentation of (e.g.) a 'disconnect session' command implies sesman will ned to talk to the relevant xrdp instance.

For now I'll add a define for this with a short comment.